---
version: "3.9"

services:
  uptime-kuma:
    image: louislam/uptime-kuma
    container_name: uptime-kuma
    volumes:
      - ${FILE1}:/app/data
      - /var/run/docker.sock:/var/run/docker.sock
    #ports:
    #  - 3001:3001
    environment:
      - "NAME_SERVERS=${NAME_SERVERS}"
    restart: unless-stopped
    security_opt:
      - no-new-privileges:true

    networks:
      - proxy

    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.${SERVICE}.entrypoints=http"
      - "traefik.http.routers.${SERVICE}.rule=Host(`${DOMAIN}`)"
      - "traefik.http.routers.portainer.middlewares=https-redirect@file"
      - "traefik.http.routers.${SERVICE}-secure.entrypoints=https"
      - "traefik.http.routers.${SERVICE}-secure.rule=Host(`${DOMAIN}`)"
      - "traefik.http.routers.${SERVICE}-secure.tls=true"
      - "traefik.http.routers.${SERVICE}-secure.service=${SERVICE}"
      - "traefik.http.services.${SERVICE}.loadbalancer.server.port=${PORT}"
      - "traefik.docker.network=proxy"

networks:
  proxy:
    external: true
